src/blockchain/contracts.py

```diff
@@ -29,28 +23,35 @@ class Contracts:
     def initialize(self, w3: Web3, abi_path='./interfaces/'):
         __initialized = True
 
+        self.lido_locator = w3.eth.contract(
+            address=LIDO_LOCATOR[WEB3_CHAIN_ID],
+            abi=load_abi(abi_path, 'LidoLocator'),
+        )
+
         self.lido = w3.eth.contract(
-            address=LIDO_CONTRACT_ADDRESSES[WEB3_CHAIN_ID],
+            address=self.lido_locator.functions.lido().call(),
             abi=load_abi(abi_path, 'Lido'),
         )
-        self.node_operator_registry = w3.eth.contract(
-            address=NODE_OPS_ADDRESSES[WEB3_CHAIN_ID],
-            abi=load_abi(abi_path, 'NodeOperatorRegistry'),
-        )
 
         self.deposit_security_module = w3.eth.contract(
-            address=DEPOSIT_SECURITY_MODULE[WEB3_CHAIN_ID],
+            address=self.lido_locator.functions.depositSecurityModule().call(),
```

Comment:
Do we need some check that returned value is an address?

Reply:
No. Response type is hardcoded in abi. So it defiantly the address.